stdlib/private/CMakeLists.txt

```diff
@@ -23,8 +23,8 @@ if(SWIFT_BUILD_SDK_OVERLAY)
 
   if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     add_subdirectory(StdlibUnittestFoundationExtras)
-    if (SWIFT_INCLUDE_TESTS)
-      add_subdirectory(SwiftReflectionTest)
-    endif()
+  endif()
+  if (SWIFT_INCLUDE_TESTS AND ((CMAKE_BUILD_TYPE MATCHES DEBUG)) OR CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
I am very against this change in the cmake. Please remove it.

Reply:
For context: @jckarter explained that some of the runtime hooks we're adding for testing purposes shouldn't be made into API. I do agree with @gottesmm we can find a way to conditionally enable those tests (say, based on a lit 'REQUIRES' directive -- see "available_features" in lit.cfg), rather than altering the build logic in this way.

More context for @augusto2112: cmake switches have a tendency to grow in complexity relatively quickly, and can sometimes make it hard for developers on different platforms to test the same thing. If we can express that a test is platform-dependent in the test itself, that's likely to be less brittle.

Reply:
The way we have done this sort of thing generally is to enable it with asserts compilers, rather than debug compilers. That way in nightly snapshots (which are release+asserts) as well as locally we still have access to these tools (which can be useful when debugging!)

Reply:
What part are you against? I need to at least place it outside the "if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")" check, and it already only compiled if SWIFT_INCLUDE_TESTS was true. And I need to make sure it only builds when the functions I added are present, or else it wouldn't compile right? How else would I do that?